website/addons/dmptool/model.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+
+
+from modularodm import fields
+
+from framework.exceptions import HTTPError
+
+from website.addons.base import (
+    AddonOAuthNodeSettingsBase, AddonOAuthUserSettingsBase, exceptions,
+)
+from website.addons.base import StorageAddonBase
+
+from website.addons.dmptool.serializer import DmptoolSerializer
+
+ROOT_FOLDER_NAME = '/'
+
+
+class DmptoolProvider(object):
+    """An alternative to `ExternalProvider` not tied to OAuth"""
+
+    name = 'Dmptool'
+    short_name = 'dmptool'
+    serializer = DmptoolSerializer
+
+    def __init__(self, account=None):
+        super(DmptoolProvider, self).__init__()
+        # provide an unauthenticated session by default
+        self.account = account
+
+    def __repr__(self):
+        return '<{name}: {status}>'.format(
+            name=self.__class__.__name__,
+            status=self.account.provider_id if self.account else 'anonymous'
+        )
+
+class AddonDmptoolUserSettings(AddonOAuthUserSettingsBase):
+    oauth_provider = DmptoolProvider
+    serializer = DmptoolSerializer
+
+class AddonDmptoolNodeSettings(StorageAddonBase, AddonOAuthNodeSettingsBase):
+    oauth_provider = DmptoolProvider
+    serializer = DmptoolSerializer
+
+    folder_id = fields.StringField(default=ROOT_FOLDER_NAME)
+    folder_name = fields.StringField(default=ROOT_FOLDER_NAME)
+    folder_path = fields.StringField(default=ROOT_FOLDER_NAME)
```

Comment:
Re the [permissions issue](https://hyp.is/fZNyiBotEee79a_5Q5bl3w/github.com/CenterForOpenScience/osf.io/pull/7030):

> Also, I don't recall if we discussed the permissions issue, but is there anyway for a user to restrict which plans appear in this addon? Right now it looks like it shows everything or nothing. I don't know if we need to work that out now, but it would be good to keep in mind.

We haven't talked about this issue of permissions.  Right now, the DMPTool addon is coded to display all the plans owned by the user at the DMPTool. Within the DMPTool, a plan owner can set a plan to be one of four visibility levels: 

* test
* private
* public
* institutional

With guidance from you, do you think it'd be hard to add functionality in Fangorn configruation (?) or the DMPTool WaterButler provider (?) to let the user mark individual plans as visible or invisible within the osf.io project?.